File: iosApp/iosApp/Screens/Parent/ParentInsightCenter.swift
import SwiftUI
import SharedLogic

struct ParentInsightCenter: View {
    let children: [SharedLogic.ChildSummary]
    let activities: [SharedLogic.ChildActivity]
    let onChildTap: (String) -> Void

    private var totalWords: Int32 {
        children.reduce(0) { $0 + $1.wordsLearned }
    }

    private var totalStars: Int32 {
        children.reduce(0) { $0 + $1.stars }
    }

    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                // KPI Grid
                LazyVGrid(columns: [GridItem(.flexible()), GridItem(.flexible())], spacing: 12) {
                    KPICard(emoji: "👶", value: "\(children.count)", label: "Hijos")
                    KPICard(emoji: "📚", value: "\(totalWords)", label: "Palabras")
                    KPICard(emoji: "⭐", value: "\(totalStars)", label: "Estrellas")
                    KPICard(emoji: "📊", value: "\(activities.count)", label: "Actividades")
                }
                .padding(.horizontal, 16)

                // Children summary
                if !children.isEmpty {
                    VStack(alignment: .leading, spacing: 10) {
                        Text("Hijos")
                            .font(.custom("DynaPuff", size: 16))
                            .fontWeight(.bold)
                            .foregroundColor(.white)
                            .padding(.horizontal, 16)

                        ForEach(children, id: \.id) { child in
                            Button { onChildTap(child.id) } label: {
                                HStack(spacing: 12) {
                                    Text("👦").font(.system(size: 20))
                                    Text(child.name)
                                        .font(.custom("DM Sans", size: 15))
                                        .fontWeight(.semibold)
                                        .foregroundColor(.white)
                                    Spacer()
                                    Text("Nivel \(child.level)")
                                        .font(.custom("DM Sans", size: 12))
                                        .foregroundColor(.white.opacity(0.6))
                                    Image(systemName: "chevron.right")
                                        .foregroundColor(.white.opacity(0.4))
                                }
                                .padding(12)
                                .background(RoundedRectangle(cornerRadius: 12).fill(.white.opacity(0.08)))
                            }
                            .buttonStyle(.plain)
                        }
                        .padding(.horizontal, 16)
                    }
                }

                // Activity timeline
                if !activities.isEmpty {
                    VStack(alignment: .leading, spacing: 10) {
                        Text("Actividad reciente")
                            .font(.custom("DynaPuff", size: 16))
                            .fontWeight(.bold)
                            .foregroundColor(.white)
                            .padding(.horizontal, 16)

                        ForEach(Array(activities.prefix(10).enumerated()), id: \.offset) { index, activity in
                            HStack(spacing: 12) {
                                VStack(spacing: 0) {
                                    Circle().fill(Color(hex: "3B7DF6")).frame(width: 8, height: 8)
                                    if index < min(9, activities.count - 1) {
                                        Rectangle().fill(.white.opacity(0.15)).frame(width: 2)
                                    }
                                }
                                .frame(width: 8)

                                VStack(alignment: .leading, spacing: 2) {
                                    Text(activity.date)
                                        .font(.custom("DM Sans", size: 11))
                                        .foregroundColor(.white.opacity(0.5))
                                    Text(activity.description_)
                                        .font(.custom("DM Sans", size: 14))
                                        .foregroundColor(.white)
                                    if !activity.detail.isEmpty {
                                        Text(activity.detail)
                                            .font(.custom("DM Sans", size: 12))
                                            .foregroundColor(.white.opacity(0.5))
                                    }
                                }
                                Spacer()
                            }
                            .padding(.vertical, 6)
                            .padding(.horizontal, 20)
                        }
                    }
                }
            }
            .padding(.bottom, 20)
        }
    }
}

private struct KPICard: View {
    let emoji: String
    let value: String
    let label: String

    var body: some View {
        VStack(spacing: 6) {
            Text(emoji).font(.system(size: 28))
            Text(value)
                .font(.custom("DynaPuff", size: 22))
                .fontWeight(.bold)
                .foregroundColor(.white)
            Text(label)
                .font(.custom("DM Sans", size: 12))
                .foregroundColor(.white.opacity(0.6))
        }
        .padding(14)
        .frame(maxWidth: .infinity)
        .background(RoundedRectangle(cornerRadius: 16).fill(.white.opacity(0.1)))
    }
}

File: sharedUI/src/commonMain/kotlin/org/alphakids/app/parent/ParentInsightCenter.kt
package org.alphakids.app.parent

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import coil3.compose.AsyncImage
import org.alphakids.app.koinInject
import org.alphakids.app.navigation.Screen
import org.alphakids.app.parent.domain.model.ChildSummary

@Composable
fun ParentInsightCenter(
    navController: NavController,
    onChildClick: (String) -> Unit,
    modifier: Modifier = Modifier,
) {
    val viewModel = remember { ParentInsightViewModel(koinInject()) }
    val state by viewModel.uiState.collectAsState()

    LazyColumn(
        modifier = modifier.fillMaxSize(),
        contentPadding = PaddingValues(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        if (state.isLoading) {
            item(key = "loading") {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 32.dp),
                    contentAlignment = Alignment.Center,
                ) {
                    CircularProgressIndicator()
                }
            }
            return@LazyColumn
        }

        if (state.error != null) {
            item(key = "error") {
                Text(
                    text = state.error ?: "",
                    color = MaterialTheme.colorScheme.error,
                    style = MaterialTheme.typography.bodyMedium,
                    modifier = Modifier.padding(vertical = 8.dp),
                )
            }
            return@LazyColumn
        }

        // KPI grid
        item(key = "kpi-grid") {
            val agg = state.aggregated
            val kpis = listOf(
                KpiItem("👶", "Hijos", "${agg.totalChildren}"),
                KpiItem("📖", "Palabras", "${agg.totalWords}"),
                KpiItem("⏱️", "Tiempo", "${agg.totalTime} min"),
                KpiItem("🪙", "Monedas", "${agg.totalCoins}"),
                KpiItem("⭐", "Estrellas", "${agg.totalStars}"),
                KpiItem("📷", "OCR", "${agg.totalOcr}"),
                KpiItem("✍️", "Deletreo", "${agg.totalSpelling}"),
                KpiItem("📊", "Nivel prom.", "${agg.avgLevel}"),
            )

            LazyVerticalGrid(
                columns = GridCells.Fixed(4),
                modifier = Modifier
                    .fillMaxWidth()
                    .height(200.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp),
                userScrollEnabled = false,
            ) {
                items(kpis) { kpi ->
                    KpiCard(kpi)
                }
            }
        }

        // Children section
        if (state.children.isNotEmpty()) {
            item(key = "children-title") {
                Text(
                    text = "👶 Tus hijos",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurface,
                    modifier = Modifier.padding(top = 4.dp),
                )
            }

            items(
                items = state.children,
                key = { it.id },
            ) { child ->
                ChildCard(
                    child = child,
                    onClick = { onChildClick(child.id) },
                )
            }
        }

        // Add child
        item(key = "add-child") {
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable {
                        navController.navigate(Screen.SetupWizard.route)
                    },
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(
                    containerColor = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f),
                ),
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(16.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text(text = "➕", style = MaterialTheme.typography.titleLarge)
                    Spacer(modifier = Modifier.width(12.dp))
                    Text(
                        text = "Agregar hijo",
                        style = MaterialTheme.typography.titleSmall,
                        fontWeight = FontWeight.Medium,
                        color = MaterialTheme.colorScheme.primary,
                    )
                }
            }
        }

        // Activity timeline
        if (state.recentActivity.isNotEmpty()) {
            item(key = "activity-title") {
                Text(
                    text = "📅 Actividad reciente",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurface,
                    modifier = Modifier.padding(top = 4.dp),
                )
            }

            items(
                items = state.recentActivity,
                key = { it.date + it.description },
            ) { activity ->
                ActivityTimelineItem(activity = activity)
            }
        }

        item(key = "bottom") {
            Spacer(modifier = Modifier.height(8.dp))
        }
    }
}

private data class KpiItem(val emoji: String, val label: String, val value: String)

@Composable
private fun KpiCard(item: KpiItem) {
    Card(
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(8.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Text(text = item.emoji, style = MaterialTheme.typography.titleMedium)
            Spacer(modifier = Modifier.height(2.dp))
            Text(
                text = item.value,
                style = MaterialTheme.typography.titleSmall,
                fontWeight = FontWeight.Bold,
                color = MaterialTheme.colorScheme.onSurface,
                textAlign = TextAlign.Center,
            )
            Text(
                text = item.label,
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }
    }
}

@Composable
private fun ChildCard(
    child: ChildSummary,
    onClick: () -> Unit,
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick),
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(12.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            val avatarUrl = "https://api.dicebear.com/10.x/adventurer-neutral/svg?seed=${child.avatarSeed}"
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .clip(CircleShape)
                    .background(MaterialTheme.colorScheme.surfaceVariant),
                contentAlignment = Alignment.Center,
            ) {
                AsyncImage(
                    model = avatarUrl,
                    contentDescription = "Avatar de ${child.name}",
                    modifier = Modifier
                        .size(44.dp)
                        .clip(CircleShape),
                    contentScale = ContentScale.Fit,
                )
            }

            Spacer(modifier = Modifier.width(12.dp))

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = child.name,
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurface,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
                Text(
                    text = "Nivel ${child.level} · ${child.rank}",
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }

            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text(text = "⭐", style = MaterialTheme.typography.titleMedium)
                Text(
                    text = "${child.stars}",
                    style = MaterialTheme.typography.labelLarge,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }
        }
    }
}

@Composable
private fun ActivityTimelineItem(
    activity: org.alphakids.app.parent.domain.model.ChildActivity,
) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(8.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(12.dp),
            verticalAlignment = Alignment.Top,
        ) {
            Box(
                modifier = Modifier
                    .padding(top = 4.dp)
                    .size(10.dp)
                    .clip(CircleShape)
                    .background(MaterialTheme.colorScheme.primary),
            )

            Spacer(modifier = Modifier.width(12.dp))

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = activity.description,
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.Medium,
                    color = MaterialTheme.colorScheme.onSurface,
                )
                Spacer(modifier = Modifier.height(2.dp))
                Row(
                    horizontalArrangement = Arrangement.SpaceBetween,
                    modifier = Modifier.fillMaxWidth(),
                ) {
                    Text(
                        text = activity.date,
                        style = MaterialTheme.typography.labelSmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                    )
                    Text(
                        text = activity.detail,
                        style = MaterialTheme.typography.labelSmall,
                        color = MaterialTheme.colorScheme.primary,
                    )
                }
            }
        }
    }
}
